### DISCUSSION TITLE
What is the state of the Test Runner API/ Coverage API and Gototest API

### DISCUSSION THREAD

**Discussion by albilu** [apache/netbeans#5942] [Q&A]
quoted from a Netbeans dev : _friend APIs are APIs which are not stable and therefore not public. If you want to use them anyway, you have to pretend to be a friend and use them. Since they can change from release to release, you have to do provide a build for every release_
...

https://github.com/apache/netbeans/tree/master/ide/gototest
https://github.com/apache/netbeans/tree/master/ide/gsf.codecoverage
https://github.com/apache/netbeans/tree/master/ide/gsf.testrunner

As those 3 APIs are very important part of a Language support module they need to be made available to allow plugin developers to work with them more smoothly.

It is now several years since those APIs are not public if i am not wrong.

I worked with them recently and didn't noticed any instability or harm to the IDE.

So what is preventing to made them public ?

**Comment by neilcsmith-net** [comment]
This discussion belongs on the dev@ mailing list. Thanks!

**Reply by albilu** [reply]
Just created a thread on the list. Thanks
